src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1409,20 +1409,28 @@ protected void AfterMembersChecks(DiagnosticBag diagnostics)
             CheckForProtectedInStaticClass(diagnostics);
             CheckForUnmatchedOperators(diagnostics);
 
+            var location = Locations[0];
             if (this.IsByRefLikeType)
             {
-                this.DeclaringCompilation.EnsureIsByRefLikeAttributeExists(diagnostics, Locations[0], modifyCompilation: true);
+                this.DeclaringCompilation.EnsureIsByRefLikeAttributeExists(diagnostics, location, modifyCompilation: true);
             }
 
             if (this.IsReadOnly)
             {
-                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(diagnostics, Locations[0], modifyCompilation: true);
+                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(diagnostics, location, modifyCompilation: true);
             }
 
-            if (this.BaseTypeNoUseSiteDiagnostics?.ContainsNullableReferenceTypes() == true ||
-                this.InterfacesNoUseSiteDiagnostics().Any(t => t.ContainsNullableReferenceTypes()))
+            var baseType = BaseTypeNoUseSiteDiagnostics;
+            var interfaces = InterfacesNoUseSiteDiagnostics();
+            if (baseType?.ContainsAnnotatedUnconstrainedTypeParameter() == true ||
+                interfaces.Any(t => t.ContainsAnnotatedUnconstrainedTypeParameter()))
             {
-                this.DeclaringCompilation.EnsureNullableAttributeExists(diagnostics, Locations[0], modifyCompilation: true);
+                TypeSymbolWithAnnotations.ReportAnnotatedUnconstrainedTypeParameter(location, diagnostics);
```

Comment:
Should we report more specific locations here too?

Reply:
Added comment.

---
In reply to: [205843155](https://github.com/dotnet/roslyn/pull/28804#discussion_r205843155) [](ancestors = 205843155)